### PR TITLE
Fix crazy peak in memory when moving selection

### LINKED ISSUE #3426: Memory usage surge when selecting/appending MultilineEntry text
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

My fyne App is just a node.exe launcher. It launcher node.exe and capture its output and write to a text entry.
Memory usage surge when selecting/appending MultilineEntry text as the following gif shows.
Memory usage keeps growing when the output of the node script is captured to the text entry. After about an hour of running, the memory usage peak at an astonishing 12G.

### How to reproduce

1. Create a multiline entry with a lot text content
2. Keep Selecting text in the entry and monitor the memory usage of the app

### Screenshots

20221124151652_rec_

image

### Example code

```go
	text := widget.NewMultiLineEntry()
	text.SetText("Press button 'Run' to Start")
	text.SetMinRowsVisible(16)

       var c *cmd.Cmd = cmd.NewCmdOptions(cmdOptions,  'node', newArgs...)
```

```go
		go func() {
			complete := c.Status().Complete
			for !complete && c.Stdout != nil || c.Stderr != nil {
				complete = c.Status().Complete
				if !complete {
					if !runButton.Disabled() {
						runButton.Disable()
					}
					if stopButton.Disabled() {
						stopButton.Enable()
					}
				}
				select {
				case line, open := <-c.Stdout:
					// OK
					if !open {
						c.Stdout = nil
						continue
					}
					log.Println(line)
					text.SetText(text.Text + "\n" + line)
				case line, open := <-c.Stderr:
					// Error
					if !open {
						c.Stderr = nil
						continue
					}
					fmt.Fprintln(os.Stderr, line)
					text.SetText(text.Text + "\n" + line)
				}
			}
			stopButton.Disable()
			runButton.Enable()
		}()
		c.Start()
		a.Lifecycle().SetOnStopped(func() {
			log.Println("Stopped")
			c.Stop()
		})
```

### Fyne version

2.2.3

### Go compiler version

1.17

### Operating system

macOS

### Operating system version

12.5.1

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by andydotxyz**
There is an underlying issue where text refresh memory is not freed

but here we are able to avoid refresh which is always good.
Fixes #3426

### Checklist:

- [ ] Tests included. <- an optimisation
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: widget/entry.go [hunk 1]
```diff
@@ -1158,17 +1158,23 @@ func (e *Entry) truncatePosition(row, col int) (int, int) {
...
-			e.CursorColumn = col
-		}
+	e.propertyLock.Lock()
 
-		if !e.selecting {
-			e.selectRow = row
-			e.selectColumn = col
-		}
-	})
+	if !rightClick || rightClick && !e.selecting {
```
Comment (Jacalz) [inline widget/entry.go:1163]: You should be able to simplify the Boolean logic slightly:
```suggestion
	if !rightClick || !e.selecting {
```
As you can see, the truth tables are the same:
image
image

**Review by Jacalz — CHANGES_REQUESTED** [review]
Nice work. Just noted a small simplification to the Boolean logic :)

File: widget/entry.go (hunk 1 above)
Comment (andydotxyz) [reply widget/entry.go:1163]: Good point. Missed that as it was just an indent change. fixing

**Review by Jacalz — APPROVED** [review]
Works well. I do not see any drastic increase when moving the cursor :+1:

**Pull request merged**